src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -26,10 +28,24 @@ class PyTestResult(TestResult):
   pass
 
 
+def parse_interpreter_constraints(python_setup, python_target_adaptors):
```

Comment:
Bikeshed time! In rules files, should we have helper functions like this appear above or below the main `@rule` and `def rules()`?

I have no preference and really don't know what's best.

Reply:
> Bikeshed time! In rules files, should we have helper functions like this appear above or below the main @rule and def rules()?

IMO, it should appear nearest to the `@rule` that it is being used by. If there were more rules in this file, they might appear above/below a static function for that reason.

But it doesn't really matter.